pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,427 @@
+            if axis.linkedView() is None:
+                axis._linkedView = weakref.ref(cvb)
+            # FROM AxisItem.linkToView
+            # connect view changes to axis changes
+            if axis.orientation in ["right", "left"]:
+                self._signalConnectionsByChart[chart.name]["cvb.sigYRangeChanged"] = cvb.sigYRangeChanged.connect(
+                    axis.linkedViewChanged)
+            elif axis.orientation in ["top", "bottom"]:
+                self._signalConnectionsByChart[chart.name]["cvb.sigXRangeChanged"] = cvb.sigXRangeChanged.connect(
+                    axis.linkedViewChanged)
+            self._signalConnectionsByChart[chart.name]["cvb.sigResized"] = cvb.sigResized.connect(
+                axis.linkedViewChanged)
+            axis_view = axis.linkedView()
+            if cvb is not axis_view:
+                # FROM ViewBox.linkView
+                # connext axis's view changes to view since axis acts just like a proxy to it
+                if axis.orientation in ["top", "bottom"]:
+                    # connect axis main view changes to view
+                    cvb.state["linkedViews"][cvb.XAxis] = weakref.ref(axis_view)
+                    # this signal is received multiple times when using mouse actions directly on the viewbox
+                    # this causes the non top layer views to scroll more than the frontmost one
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigXRangeChanged"] = axis_view.sigXRangeChanged.connect(
+                        cvb.linkedXChanged)
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigResized"] = axis_view.sigResized.connect(
+                        cvb.linkedXChanged)
+                    # disable autorange on manual movements
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigXRangeChangedManually"] = axis_view.sigXRangeChangedManually.connect(
+                        lambda mask: self.disableAxisAutoRange(axis_name))
+                elif axis.orientation in ["right", "left"]:
+                    # connect axis main view changes to view
+                    cvb.state["linkedViews"][cvb.YAxis] = weakref.ref(axis_view)
+                    # this signal is received multiple times when using mouse actions directly on the viewbox
+                    # this causes the non top layer views to scroll more than the frontmost one
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigYRangeChanged"] = axis_view.sigYRangeChanged.connect(
+                        cvb.linkedYChanged)
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigResized"] = axis_view.sigResized.connect(
+                        cvb.linkedYChanged)
+                    # disable autorange on manual movements
+                    self._signalConnectionsByChart[chart.name]["axis_view.sigYRangeChangedManually"] = axis_view.sigYRangeChangedManually.connect(
```

Comment:
@outofculture may chime in here, but we've found that signals that are connected to lambda methods that use `self` to be problematic (see https://github.com/pyqtgraph/pyqtgraph/pull/1598 ).  To work around this we definitely had to make some code substantially more verbose.

Reply:
i can fix the lambda that actually doesn't invoke a self.a_function but for the others i need to pass the axis_name that is a variable from the running scope and is not retrievable from the parameters of the signal at least at a first glance, any suggestions?

Reply:
The other `signal.connect(lambda: ...)` that don't call `self` are fine; from what I remember (and again, outofculture may have something to add here as he did the bulk of the investigation here), the `lambda` function referencing `self` was the problem, not just the `lambda` function being connected to a signal.

Reply:
is using lambda: class.function(self, arguments) fine? being class = type(self)

Reply:
added workaround